Libs/RaptureCore/src/main/java/rapture/blob/google/GoogleBlobStore.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2011-2016 Incapture Technologies LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package rapture.blob.google;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.google.cloud.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+
+import rapture.blob.BaseBlobStore;
+import rapture.blob.BlobStore;
+import rapture.common.CallingContext;
+import rapture.common.RaptureURI;
+
+/**
+ * GoogleBlobStore uses the Google Cloud Storage to implement a BlobStore Google has the concept of Buckets which store blobs. At first pass it seems logical to
+ * map Repositories to Buckets; will see if that holds up. The downside of this is that buckets are globally named, so we'll need a prefix that guarantees all
+ * our repositories are our own. Bucket names containing dots require verification which we will need to do at some point.
+ * 
+ * @author Dave Tong
+ * 
+ */
+public class GoogleBlobStore extends BaseBlobStore implements BlobStore {
+
+    private static Logger logger = Logger.getLogger(GoogleBlobStore.class);
+
+    private static final String INCAPTURE = "incapture_"; // MUST BE LOWER CASE
```

Comment:
My understanding is that the bucket name needs to be globally unique as all buckets share the same namespace. Google encourage this by allowing DNS type naming. However in order to do this you have to verify to Google that you own the domain name, and I haven't done that yet.